source/adapters/cuda/image.cpp

```diff
@@ -455,32 +455,40 @@ UR_APIEXPORT ur_result_t UR_APICALL urBindlessImagesImageAllocateExp(
 
   // Allocate a cuArray
   if (pImageDesc->numMipLevel == 1) {
-    CUarray ImageArray;
+    CUarray ImageArray{};
 
     try {
       UR_CHECK_ERROR(cuArray3DCreate(&ImageArray, &array_desc));
       *phImageMem = (ur_exp_image_mem_native_handle_t)ImageArray;
     } catch (ur_result_t Err) {
-      cuArrayDestroy(ImageArray);
+      if (ImageArray) {
```

Comment:
I'm sure it's OK to do as you suggest but just in case `CUarray` is not aliased to a ptr type. I think this should be applied for all of the checks

```suggestion
      if (ImageArray != CUarray{}) {
```

Reply:
Good point! Applied. With the hip adapter the handle is declared as a pointer but I added it nonetheless for completeness if the code is ever copied for use where that safety isn't guaranteed.